src/web3py/extensions/tx_utils.py

```diff
@@ -42,6 +42,9 @@ def _check_transaction(transaction, params: TxParams) -> bool:
         except ContractLogicError as error:
             logger.warning({"msg": "Transaction reverted.", "error": str(error)})
             return False
+        except ValueError as error:
+            logger.error({"msg": "Not enough funds.", "error": str(error)})
```

Comment:
is this the only possible cause of the ValueError?

Reply:
We will see details in "error" and if it is another exception we will see it
But for now as I know yes